sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/StateStoreSuite.scala

```diff
@@ -814,11 +818,53 @@ class StateStoreSuite extends StateStoreSuiteBase[HDFSBackedStateStoreProvider]
   }
 }
 
+class StateStoreCompatibleSuite extends StreamTest with StateStoreCodecsTest {
+  testWithAllCodec("SPARK-33263: Recovery from checkpoint before codec config introduced") {
+    val resourceUri = this.getClass.getResource(
+      "/structured-streaming/checkpoint-version-3.0.0-streaming-statestore-codec/").toURI
+    val checkpointDir = Utils.createTempDir().getCanonicalFile
+    FileUtils.copyDirectory(new File(resourceUri), checkpointDir)
+
+    import testImplicits._
+
+    val inputData = MemoryStream[Int]
+    val aggregated = inputData.toDF().groupBy("value").agg(count("*"))
+    inputData.addData(1, 2, 3)
+
+    /*
+      Note: The checkpoint was generated using the following input in Spark version 3.0.0:
+      AddData(inputData, 1, 2, 3)
+    */
```

Comment:
nit. Shall we adjust a little?
- https://github.com/databricks/scala-style-guide#documentation-style